extension/encoding/awslogsencodingextension/internal/unmarshaler/cloudtraillogs/cloudtraillogs.go

```diff
@@ -0,0 +1,283 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cloudtraillogs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/klauspost/compress/gzip"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/otel/semconv/v1.27.0"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/internal/metadata"
+)
+
+type CloudTrailLogsUnmarshaler struct {
+	buildInfo component.BuildInfo
+	// Pool the gzip readers, which are expensive to create.
+	gzipPool sync.Pool
+}
+
+var _ plog.Unmarshaler = (*CloudTrailLogsUnmarshaler)(nil)
+
+// CloudTrailRecord represents a CloudTrail log record
+// There is no builtin CloudTrailRecord we can leverage like in S3
+// So we build our own
+type CloudTrailRecord struct {
+	EventVersion                 string         `json:"eventVersion"`
+	EventTime                    string         `json:"eventTime"`
+	EventSource                  string         `json:"eventSource"`
+	EventName                    string         `json:"eventName"`
+	AwsRegion                    string         `json:"awsRegion"`
+	SourceIPAddress              string         `json:"sourceIPAddress"`
+	UserAgent                    string         `json:"userAgent"`
+	RequestID                    string         `json:"requestID"`
+	EventID                      string         `json:"eventID"`
+	EventType                    string         `json:"eventType"`
+	EventCategory                string         `json:"eventCategory"`
+	RecipientAccountID           string         `json:"recipientAccountId"`
+	UserIdentity                 map[string]any `json:"userIdentity"`
```

Comment:
I think we should try to remove any use of `any`. They are quite expensive in performance. I see, for example, for user identity we only care about `principalId`, `userName` and `arn`, and they are all strings, so maybe we could even replace this with

```
UserIdentity {
  PrincipalID string `mapstructure: principalId`
  ...
} `json:"userIdentity"`
```

What do you think?

And then the same would apply for any other `any`s.